subprojects/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/TestExecutionBuildConfigurationAction.java

```diff
@@ -57,6 +58,14 @@ private void configureTestTasks(Set<Test> allTestTasksToRun) {
             task.setIgnoreFailures(true);
             task.getFilter().setFailOnNoMatchingTests(false);
             task.getOutputs().upToDateWhen(Specs.SATISFIES_NONE);
+            InternalDebugOptions debugOptions = testExecutionRequest.getDebugOptions();
+            if (debugOptions.isDebugMode()) {
+                task.setDebug(true);
+                task.setDebugOptions(debugOptions.getPort());
+                // run all tests on the same JVM
+                task.setForkEvery(null);
```

Comment:
❌ This should be handled in the test task, not here.

Reply:
Fixed.